query-engine/connectors/sql-query-connector/src/error.rs

```diff
@@ -5,13 +5,19 @@ use std::{any::Any, string::FromUtf8Error};
 use thiserror::Error;
 use user_facing_errors::query_engine::DatabaseConstraint;
 
-pub(crate) enum RawError {
-    IncorrectNumberOfParameters {
-        expected: usize,
-        actual: usize,
-    },
+#[cfg(not(target_arch = "wasm32"))]
+use quaint::error::NativeErrorKind;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(crate) enum NativeRawError {
+    IncorrectNumberOfParameters { expected: usize, actual: usize },
```

Comment:
Same here, I feel like this error is something we should have for driver adapters, even if we are not doing it right now.

Reply:
For this one, you're probably right. I'm afraid `driver-adapters` are currently susceptible to https://github.com/prisma/prisma/issues/5077.

It should be noted that, currently, this error is only Postgres/CockroachDB-specific: https://github.com/prisma/prisma-engines/blob/0ac27a9109860a23c4a20cdd41b97fc61f8ab172/quaint/src/connector/postgres/native/mod.rs#L373-L380.

Reply:
I'm actually unable to trigger this error.
I've tried things like:

```typescript
  await prisma.user.create({
    data: {
      name: 'jkomyno',
      id: '1'
    },
  })

  const data1 = await prisma.$queryRawUnsafe(`SELECT * FROM "User"`, "extra param");
  console.log('data1') // [ { id: '1', name: 'jkomyno' } ]

  const data2 = await prisma.$queryRaw(Prisma.sql`SELECT *, ${"extra param"} FROM "User"`);
  console.log('data2') // [ { id: '1', name: 'jkomyno', '?column?': 'extra param' } ]
```

with the native Postgres driver, against `postgres:10`, `postgres:16`, `neon`, and in every case Prisma resolved with a successful result.

Reply:
Using `driver-adapters` (with `pg`), on the other hand, results in the following `RawQueryError`:

```
Raw query failed. Code: `08P01`. Message: `ERROR: bind message supplies 1 parameters, but prepared statement "" requires 0`
    at si.handleRequestError (/Users/jkomyno/work/prisma/prisma/sandbox/driver-adapters/node_modules/.prisma/client/runtime/library.js:125:6817)
    at si.handleAndLogRequestError (/Users/jkomyno/work/prisma/prisma/sandbox/driver-adapters/node_modules/.prisma/client/runtime/library.js:125:6151)
    at si.request (/Users/jkomyno/work/prisma/prisma/sandbox/driver-adapters/node_modules/.prisma/client/runtime/library.js:125:5859)
    at async l (/Users/jkomyno/work/prisma/prisma/sandbox/driver-adapters/node_modules/.prisma/client/runtime/library.js:130:9805)
    at main (/Users/jkomyno/work/prisma/prisma/sandbox/driver-adapters/src/pg.ts:27:17) {
  code: 'P2010',
  clientVersion: '0.0.0',
  meta: {
    code: '08P01',
    message: 'ERROR: bind message supplies 1 parameters, but prepared statement "" requires 0'
  }
}
```

Reply:
I've brought `IncorrectNumberOfParameters` back.